openfga_sdk/client/client.py

```diff
@@ -629,12 +643,124 @@ async def batch_check(
+            self._single_client_batch_check(request, sem, options) for request in body
         ]
         batch_check_response = await asyncio.gather(*batch_check_coros)
 
         return batch_check_response
 
+    async def _single_batch_check(
+        self,
+        body: BatchCheckRequest,
+        semaphore: asyncio.Semaphore,
+        options: dict[str, str] = None,
+    ):
+        """
+        Run a single BatchCheck request
+        :param body - list[ClientCheckRequest] defining check request
+        :param authorization_model_id(options) - Overrides the authorization model id in the configuration
+        """
+        await semaphore.acquire()
+        try:
+            kwargs = options_to_kwargs(options)
+            api_response = await self._api.batch_check(body, **kwargs)
+            return api_response
+        except Exception as err:
+            raise err
+        finally:
+            semaphore.release()
+
+    async def batch_check(self, body: ClientBatchCheckRequest, options=None):
+        """
+        Run a batchcheck request
+        :param body - BatchCheck request
+        :param authorization_model_id(options) - Overrides the authorization model id in the configuration
+        :param max_parallel_requests(options) - Max number of requests to issue in parallel. Defaults to 10
+        :param max_batch_size(options) - Max number of checks to include in a request. Defaults to 50
+        :param header(options) - Custom headers to send alongside the request
+        :param retryParams(options) - Override the retry parameters for this request
+        :param retryParams.maxRetry(options) - Override the max number of retries on each API request
+        :param retryParams.minWaitInMs(options) - Override the minimum wait before a retry is initiated
+        """
+        options = set_heading_if_not_set(
+            options, CLIENT_BULK_REQUEST_ID_HEADER, str(uuid.uuid4())
+        )
+
+        max_parallel_requests = 10
+        if options is not None and "max_parallel_requests" in options:
+            if (
+                isinstance(options["max_parallel_requests"], str)
+                and options["max_parallel_requests"].isdigit()
+            ):
+                max_parallel_requests = int(options["max_parallel_requests"])
+            elif isinstance(options["max_parallel_requests"], int):
+                max_parallel_requests = options["max_parallel_requests"]
+
+        max_batch_size = 50
+        if options is not None and "max_batch_size" in options:
+            if (
+                isinstance(options["max_batch_size"], str)
+                and options["max_batch_size"].isdigit()
+            ):
+                max_batch_size = int(options["max_batch_size"])
+            elif isinstance(options["max_batch_size"], int):
+                max_batch_size = options["max_batch_size"]
+
+        check_to_id: dict[str, ClientBatchCheckItem] = {}
```

Comment:
nit: isn't this dict an id to check mapping? so `id_to_check`?

Reply:
Yeah good callout, will fix

Reply:
Handled in 69b65dff87e158ba35d78f191b64b8ea8e463f60